src/main/java/blusunrize/immersiveengineering/api/excavator/MineralVein.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * BluSunrize
+ * Copyright (c) 2020
+ *
+ * This code is licensed under "Blu's License of Common Sense"
+ * Details can be found in the license file in the root folder of this project
+ *
+ */
+
+package blusunrize.immersiveengineering.api.excavator;
+
+import blusunrize.immersiveengineering.common.IESaveData;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.ResourceLocationException;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ColumnPos;
+
+import javax.annotation.Nullable;
+
+public class MineralVein
+{
+	private final ColumnPos pos;
+	private final MineralMix mineral;
+	private final int radius;
+	private int depletion;
+
+	public MineralVein(ColumnPos pos, MineralMix mineral, int radius)
+	{
+		this.pos = pos;
+		this.mineral = mineral;
+		this.radius = radius;
+	}
+
+	public ColumnPos getPos()
+	{
+		return pos;
+	}
+
+	public MineralMix getMineral()
+	{
+		return mineral;
+	}
+
+	public int getRadius()
+	{
+		return radius;
+	}
+
+	// exponential growth of the fail chance depending on the distance from the center of the vein
```

Comment:
That still isn't exponential, and it never was. Maybe just remove the comment?

Reply:
I'm bad at maths! D: